timeseries/src/autogluon/timeseries/models/tabular/tabular_model.py

```diff
@@ -0,0 +1,198 @@
+            selected_slice = slice(None, None)
+        else:
+            selected_slice = slice(-last_k_values, None)
+
+        # TODO Parrallelize the code
+        dataframe_per_item = []
+        for item_id in data.item_ids:
+            series = data.loc[item_id][self.target]
+            time_feature_columns = {
+                feature.__class__.__name__: feature(series.index[selected_slice]) for feature in self._time_features
+            }
+            lag_columns = {f"lag_{idx}": series.shift(idx).values.ravel()[selected_slice] for idx in self._lag_indices}
+            columns = {**time_feature_columns, **lag_columns, "target": series.values.ravel()[selected_slice]}
+            df = pd.DataFrame(columns, index=series.index[selected_slice])
+            dataframe_per_item.append(df.iloc[selected_slice])
+
+        return pd.concat(dataframe_per_item, ignore_index=True)
+
+    def _generate_features_from_freq(self, freq: str):
+        self._lag_indices = get_lags_for_frequency(freq)
+        self._time_features = time_features_from_frequency_str(freq)
+
+    def _fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: int = None,
+        **kwargs,
+    ) -> None:
+        self._check_fit_params()
+        if self.tabular_predictor._learner.is_fit:
+            raise AssertionError(f"{self.name} predictor has already been fit!")
+        self._generate_features_from_freq(train_data.freq)
+
+        train_data, _ = self._normalize_targets(train_data)
+        train_df = self._get_features_dataframe(train_data)
+        # Remove features that are completely missing in the training set
+        train_df.dropna(axis=1, how="all", inplace=True)
+        self._available_features = train_df.columns
+
+        if val_data is not None:
+            if val_data.freq != train_data.freq:
+                raise ValueError(
+                    f"train_data and val_data must have the same freq (received {train_data.freq} and {val_data.freq})"
+                )
+            val_data, _ = self._normalize_targets(val_data)
+            val_df = self._get_features_dataframe(val_data, last_k_values=self.prediction_length)
+            val_df = val_df[self._available_features]
+        else:
+            logger.warning(
+                f"No val_data was provided to {self.name}. "
+                "TabularPredictor will generate a validation set without respecting the temporal ordering."
+            )
+            val_df = None
+
+        # TODO: Other presets for TabularPredictor?
+        # TODO: Speed up prediction for Quantile XT/RF models
+        # TODO: Add catboost with MultiQuantile loss (after catboost v1.1)
+        # TODO: Other tabular models in quantile? (LightGBM?)
+        self.tabular_predictor.fit(
+            train_data=train_df,
+            tuning_data=val_df,
+            time_limit=time_limit,
+            excluded_model_types=["XT", "RF"],
```

Comment:
shouldn't we be allowlisting models instead of excluding them? what if Tabular introduced a new non-quantile compliant model? Can we do this through `hyperparameters`?

Reply:
My point is that we should be allowlisting the models to use, not denylisting them. The scenario is, what if Tabular introduced `XTWORSE` algorithm that's even worse than XT? This would immediately lead to all of AutoGluonTabularModel breaking.

Reply:
Got it!